receiver/dotnetdiagnosticsreceiver/metrics/converter.go

```diff
@@ -0,0 +1,70 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver/dotnet"
+)
+
+func rawMetricsToPdata(rawMetrics []dotnet.Metric) pdata.Metrics {
+	pdm := pdata.NewMetrics()
+	rms := pdm.ResourceMetrics()
+	rms.Resize(1)
+	rm := rms.At(0)
+	rm.Resource().Attributes()
+	ilms := rm.InstrumentationLibraryMetrics()
+	ilms.Resize(1)
+	ilm := ilms.At(0)
+	ms := ilm.Metrics()
+	ms.Resize(len(rawMetrics))
+	for i := 0; i < len(rawMetrics); i++ {
+		rawMetricToPdata(rawMetrics[i], ms.At(i))
+	}
+	return pdm
+}
+
+func rawMetricToPdata(dm dotnet.Metric, pdm pdata.Metric) pdata.Metric {
```

Comment:
Should I set startTime and timeStamp for these datapoints? 👇

Reply:
Thanks Tigran. For now, I have set the start time to the start time of the receiver. I'm not sure that we'll be able to get the start time of the dotnet process -- will investigate further. In any case, I'm curious about how much difference using the dotnet process start time vs receiver start time will make since we're reporting deltas.

Reply:
Ah, I misunderstood, I thought it is cumulative.
If it is delta then you need to set the start time to the start of the collection period, not start of the process. You can treat each previous timestamp is the start of the new collection period. See delta temporarily explanation here: https://docs.google.com/document/d/1rncVUp_nzAhA39trS1hUFinIlO01jf7KBbuIRhqMI7Y/edit#heading=h.16xj0a9h292v

Reply:
Ok, cool, thanks for the links. I've set the start time to the previous collection time, so the time difference is approximately the scrape interval.